### PR TITLE
overscroll, fastswipe, maxdistance

### PR DESCRIPTION
**Pull request by braco**
Adds new features I needed, since the core [SwipeableRow](https://github.com/facebook/react-native/tree/master/Libraries/Experimental/SwipeableRow) seems stalled

New props / features:

* maxLeftSwipeDistance: number
* maxRightSwipeDistance: number
* elasticOverscroll: bool
* fastSwipeVelocity: number
* onFastSwipeLeft: function
* onFastSwipeRight: function

Enables the following features:

**Maximum horizontal drag distance with elastic overscroll that snaps back on release**

This should probably be used in concert with openValue:
```
<SwipeRow leftOpenValue={200} maxLeftSwipeDistance={200} elasticOverscroll={true} />
```

And can provide a more intuitive disabled direction that allows for some stretchy overscroll:
```
<SwipeRow maxRightSwipeDistance={-10} elasticOverscroll={true} />
```

**Fast swiping (a quick directional swipe could perform the default button action)**

```
<SwipeRow onFastSwipeRight={this.deleteRow} />
```

### CODE REVIEW

File: components/SwipeRow.js [hunk 1]
```diff
@@ -101,8 +104,37 @@ class SwipeRow extends Component {
...
+			if (this.props.onFastSwipeLeft && newDX < 0 && vx >= this.props.fastSwipeVelocity) {
+				this.props.onFastSwipeLeft()
+			}
+			else if (this.props.onFastSwipeRight && newDX > 0 && vx >= this.props.fastSwipeVelocity) {
+				this.props.onFastSwipeRight()
+			}
+
+			// elasticOverscroll
+			// let scale = 200;
+			// newDX = scale * Math.log(newDX + scale) - scale * Math.log(scale);
```
Comment (jemise111) [inline components/SwipeRow.js]: Were these comments left here for a reason? If not please remove

**Comment by jemise111** [comment]
@braco this looks great thanks for doing this! If you could do a couple things I'd be happy to merge it in:

1. Add the other 3 props to SwipeRow propTypes
2. Add all 5 props to SwipeListView and pass them to the SwipeRow here: https://github.com/jemise111/react-native-swipe-list-view/blob/master/components/SwipeListView.js#L93
3. Use these new props in one of the SwipeRows or SwipeListViews in example.js

Thanks!

**Comment by VinceBT** [comment]
I am interested in this PR, any merge soon ?

**Comment by basdvries** [comment]
Any updates on when this will/can be merged?

**Comment by jemise111** [comment]
I'd love it if @braco could address the things I commented on earlier. And take a look at the issue @VinceBT found. I'll try to take a look myself asap

**Comment by braco** [comment]
See https://github.com/jemise111/react-native-swipe-list-view/pull/58

**Pull request closed without merging**